src/annotator/anchoring/test/types-test.js

```diff
@@ -19,43 +21,284 @@ describe('types', function () {
+      fakeToTextPosition = sinon.stub();
+      $imports.$mock({
+        'dom-anchor-text-quote': {
+          fromRange: fakeQuoteFromRange,
+          toRange: fakeQuoteToRange,
+          toTextPosition: fakeToTextPosition,
+        },
+      });
+    });
+
+    afterEach(() => {
+      $imports.$restore();
+    });
+
+    function createTextQuoteAnchor() {
+      return new TextQuoteAnchor(container, 'Liberty', {
+        prefix: 'a new nation, conceived in ',
+        suffix: ', and dedicated to the proposition that',
+      });
+    }
+
+    describe('#fromRange', () => {
+      it('returns a range object', () => {
+        fakeQuoteFromRange.returns({
+          prefix: 'Four score and ',
+          suffix: 'brought forth on this continent',
+        });
+        const anchor = TextQuoteAnchor.fromRange(container, new Range());
+        assert.called(fakeQuoteFromRange);
+        assert.instanceOf(anchor, TextQuoteAnchor);
+      });
+    });
+
+    describe('#fromSelector', () => {
+      it('returns a TextQuoteSelector', () => {
+        const anchor = TextQuoteAnchor.fromSelector(container, {
+          exact: 'Liberty',
+          prefix: 'a new nation, conceived in ',
+          suffix: ', and dedicated to the proposition that',
+        });
+        assert.instanceOf(anchor, TextQuoteAnchor);
+      });
+    });
+
+    describe('#toSelector', () => {
+      it('returns a TextQuoteAnchor instance', () => {
+        const anchor = createTextQuoteAnchor();
+        assert.deepEqual(anchor.toSelector(), {
+          type: 'TextQuoteSelector',
+          exact: 'Liberty',
+          prefix: 'a new nation, conceived in ',
+          suffix: ', and dedicated to the proposition that',
+        });
+      });
+    });
+
+    describe('#toRange', () => {
+      it('returns a valid DOM Range', () => {
+        $imports.$restore({
+          'dom-anchor-text-quote': {
+            toRange: true,
+          },
+        });
         const quoteAnchor = new TextQuoteAnchor(container, 'Liberty');
         const range = quoteAnchor.toRange();
         assert.instanceOf(range, Range);
         assert.equal(range.toString(), 'Liberty');
       });
 
-      it('throws if the quote is not found', function () {
+      it('throws if the quote is not found', () => {
+        $imports.$restore({
+          'dom-anchor-text-quote': {
+            toRange: true,
+          },
+        });
         const quoteAnchor = new TextQuoteAnchor(
           container,
           'five score and nine years ago'
         );
-        assert.throws(function () {
+        assert.throws(() => {
           quoteAnchor.toRange();
         });
       });
     });
 
-    describe('#toPositionAnchor', function () {
-      it('returns a TextPositionAnchor', function () {
+    describe('#toPositionAnchor', () => {
+      it('returns a TextPositionAnchor', () => {
+        $imports.$restore({
+          'dom-anchor-text-quote': {
+            toTextPosition: true,
+          },
```

Comment:
I see that the `$imports.$restore` is added here to make the test behave the same as it did before the PR. For a future reader it might be unclear why the `dom-anchor-text-quote` and `dom-anchor-text-position` libraries are mocked in some tests but not others.

Since these modules are third-party code there is some value in having tests that don't mock them because it helps to catch unexpected changes in the third-party library. We could leave that to the integration tests, although debugging failures there will be harder because there is a larger surface area to debug.

I'm OK with leaving this as-is for the moment.

Reply:
an integration test section is probably important here. I'll work on that.